UnitySDK/Assets/ML-Agents/Examples/3DBall/Scripts/Ball3DAgent.cs

```diff
@@ -63,7 +68,20 @@ public override void AgentReset()
         ballRb.velocity = new Vector3(0f, 0f, 0f);
         ball.transform.position = new Vector3(Random.Range(-1.5f, 1.5f), 4f, Random.Range(-1.5f, 1.5f))
                                       + gameObject.transform.position;
+        //Reset the parameters when the Agent is reset.
+        SetResetParameters();
+    }
 
+    public void SetBall()
+    {
+        //Set the attributes of the ball by fetching the information from the academy
+        ballRb.mass = resetParams["mass"];
+        scale = resetParams["scale"];
+        ball.transform.localScale = new Vector3(scale, scale, scale);
     }
 
+    public void SetResetParameters()
```

Comment:
This method is just calling the SetBall method. I think you should call set ball directly.

Reply:
I wanted to set precedence on how its implemented. This template of implementation is suitable when more reset parameters are introduced. Currently, all environments with reset parameter have their resets implemented in academy_reset instead of agent_reset (with the exception of WallJump). However it still does resemble the idea of the template I am using here.

Reply:
(Following up with our conversation yesterday) The rationale for seems OK for this - I'm inclined to leave it like this so that there is a consistent method call across all the examples.